### PR TITLE
JNI support for ReadOptions::iterate_lower_bound

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes: #4401

### CODE REVIEW

**Comment by moozzyk** [comment]
@seancarroll - not sure. The way I read this comment is that "There is no lower bound on the iterator." applies to the forward iterator discussed in the comment. In that sense this information is still correct. On the other hand, because it is a forward iterator, having a lower bound should not matter so maybe this sentence is not needed at all?

**Comment by seancarroll** [comment]
Ah yes that makes sense.

File: java/src/test/java/org/rocksdb/ReadOptionsTest.java [hunk 1]
```diff
@@ -144,6 +144,22 @@ public void iterateUpperBoundNull() {
...
+
+  @Test
+  public void iterateLowerBoundNull() {
+    try (final ReadOptions opt = new ReadOptions()) {
+      assertThat(opt.iterateLowerBound()).isNull();
+    }
+  }
+
   @Test
   public void copyConstructor() {
```
Comment (sagar0) [inline java/src/test/java/org/rocksdb/ReadOptionsTest.java]: Can you update this copyConstructor test as well with iterateLowerBound ? Also add fail* tests (see `failSetIterateUpperBoundUninitialized` and `failIterateUpperBoundUninitialized`).

Comment (moozzyk) [reply java/src/test/java/org/rocksdb/ReadOptionsTest.java]: Added missing test cases.

**Pull request closed without merging**